Add apimodules to test filter

reutshal opened the pull request:
<!-- REMINDER: THIS IS A PUBLIC REPO DO NOT POST HERE SECRETS/SENSITIVE DATA -->

## Status
- [X] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/23141

## Description
A few sentences describing the overall goals of the pull request's commits.

reutshal commented:
@ronykoz This is an example of a build in which I changed the CSVFeedApiModule , you can see under test-filter the integrations that was added: https://circleci.com/gh/demisto/content/54339#build-parameters/containers/0

yaakovi commented on dev-requirements-py3.txt:
```diff
@@ -1,7 +1,7 @@
 bandit==1.6.2
 demisto-py==2.0.11
-demisto-sdk==1.0.8
+git+https://github.com/demisto/demisto-sdk.git@add-apiModules_to_id_set
```
don't forget to revert

reutshal commented:
@ronykoz Tests that were made:
* Added a new ApiModule + imported in integration and checked it appears in the tests.
* modified an ApiModule and checked all relevant integrations appears in the tests.

ronykoz commented on Tests/scripts/collect_tests_and_content_packs.py:
```diff
@@ -544,11 +544,25 @@ def collect_content_packs_to_install(id_set: Dict, integration_ids: set, playboo
+    integration_to_version = {}
+    integration_ids_to_test = set([])
+    for integration in integration_set:
+        if list(integration.values())[0].get('api_modules', '') in changed_api_modules:
```
let's save this one as arg `list(integration.values())[0]`

yaakovi commented:
@reutshal unit tests are failing
